src/Reddit/Reddit.ts

```diff
@@ -1,46 +1,328 @@
-import { ExecutionContext } from "ava";
+import {ExecutionContext} from "ava";
+import {setStyles, selectAll, select} from "./utils";
 
 /*
  * LipSurf plugin for Reddit.com
  */
 /// <reference types="@lipsurf/types/extension"/>
 declare const PluginBase: IPluginBase;
 
+type Maybe<T> = T | null
+
 const thingAttr = `${PluginBase.util.getNoCollisionUniqueAttr()}-thing`;
 const COMMENTS_REGX = /reddit.com\/r\/[^\/]*\/comments\//;
+const REDDIT_REGX = /(old\.)?reddit\.com(\/)?$/;
+
+let isOldReddit = false;
+let scrollContainer: Maybe<ParentNode> = null;
+let observer: Maybe<MutationObserver> = null;
+let posts: Maybe<NodeListOf<HTMLElement>> = null;
+let index = 0;
+let isDOMLoaded = false;
+
+const reddit = {
+  old: {
+    post: {
+      thing: "#siteTable>.thing",
+      title: "a.title",
+      expandBtn: ".expando-button",
+      commentarea: ".commentarea"
+    },
+    comments: {
+      select: "a.comments",
+      expandBtn: ".expando-button",
+      comment: {
+        select: ".comment",
+        expandBtn: "a.expand",
+      }
+    },
+    special: {
+      collapsed: ".collapsed",
+      expanded: ".expanded",
+      notCollapsed: ":not(.collapsed)"
+    },
+    vote: {
+      btn: "#siteTable *[role='button']",
+      up: ".arrow.up:not(.upmod)",
+      down: ".arrow.down:not(.downmod)",
+      upmod: ".arrow.upmod",
+      downmod: ".arrow.downmod",
+    }
+  },
+  latest: {
+    post: {
+      thing: ".Post",
+    },
+    comments: {
+      select: "a[data-click-id='comments']",
+      threadline: ".threadline",
+      comment: {
+        select: ".Comment",
+        expandBtn: ".icon-expand",
+      }
+    },
+    vote: {
+      btn: ".voteButton",
+      up: ".voteButton[aria-label='upvote']",
+      down: ".voteButton[aria-label='downvote']",
+      pressed: ".voteButton[aria-pressed='true']",
+      unpressed: ".voteButton[aria-pressed='false']",
+    }
+  }
+};
 
-function thingAtIndex(i: number) {
-  return `#siteTable>div.thing[${thingAttr}="${i}"]`;
+function thingAtIndex(i: number): string {
+  if (isOldReddit) {
+    return `${reddit.old.post.thing}[${thingAttr}="${i}"]`;
+  } else {
+    return `${reddit.latest.post.thing}[${thingAttr}="${i}"]`;
+  }
 }
 
 function clickIfExists(selector: string) {
-  const el = document.querySelector<HTMLElement>(selector);
+  const el = select<HTMLElement>(selector);
   if (el) el.click();
 }
 
+function clickIfDisplayed(el: HTMLElement) {
+  if (parseFloat(getComputedStyle(el).width)) {
```

Comment:
because this is a different case, reddit hides elements from the screen, specifically the expand button, using the width: 0px style, PluginBase.util.isVisible will react to it since the element will be in the visible part of the screen but with a width of 0px

Reply:
@wiseadme ok that's fine.